src/components/ContentPanel.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { FunctionComponent } from 'react';
+import { ColumnLayout, ComponentPair, RowLayout } from './inputs/layouts';
+
+import styled from 'styled-components';
+
+export interface TabData {
+    /** title of tab */
+    title: string;
+    /** tab id, abbreviation to be displayed in side bar */
+    id?: string;
+    /** optional over-ride for tab button text colour */
+    idColour?: string;
+    /** if provided, replaces tab button text with image */
+    tabImg?: string;
+    /** alt text for replaced image */
+    imgAlt?: string;
+    /** optional background colour for tab button */
+    tabColour?: string;
+    /** components to be displayed under tab */
+    elements: ElementData[];    
+}
+
+export interface ElementData {
+    label?: string,
+    labelColour?: string,
+    labelFont?: string,
+    columns: {
+        components: ComponentPair[]
+    }[];
+}
+
+export interface ContentPanelProps
+{
+    /** panel width, defaults to '400px' */
+    panelWidth?: string;
+    /** Information to be rendered in content panel */
+    tabData: TabData;
+}
+
+interface ContainerProps {
+    panelWidth?: string;
+}
+
+const Container = styled.div<ContainerProps>`
```

Comment:
Naive question: do we have a plan for theming these styled components? Could we consider these css as default settings that could be overridden by another set of styles?

Reply:
yeah theming is something we will be adding once we get the basic inputs in